### PR TITLE
Update code for display movies by getting data from API

### PR DESCRIPTION
**Pull request by jinety**

### CODE REVIEW

File: big-practice/src/js/pages/dashboard.page.js [hunk 1]
```diff
@@ -1,114 +1,115 @@
-import { Validation } from '../validates/form.validate';
-import { EMPTY_TEXT, MOVIES_API, MESSAGES } from '../constants/message.constant';
+import { formValidate } from '../validates/form.validate';
+import { MOVIES_API } from '../constants/url-api.constant';
+import { EMPTY_TEXT, MESSAGES } from '../constants/message.constant';
 import {
-  getApi, postApi, putApi, deleteApi,
+  get, post, put, remove,
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: ApiService is class, why you need to import get/post/put/remove?

File: big-practice/src/js/pages/dashboard.page.js [hunk 2]
```diff
@@ -1,114 +1,115 @@
...
 
-  showModal(modalWarning);
-  modalWarningDeleteBtn.setAttribute('data-id', movieId);
-};
+  /**
+   * Clean error message
+   *
+   * @param {HtmlInputElement} element - Element input
+   */
+  cleanErrorMessage(element) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Should be function in Document Helper?

File: big-practice/src/js/pages/dashboard.page.js [hunk 3]
```diff
@@ -1,114 +1,115 @@
...
- */
-const cleanErrorMessage = (element) => {
-  showErrorMessage(element, EMPTY_TEXT);
-};
+  /**
+   * Hide element
+   *
+   * @param {element} element - Element
+   */
+  hideElement(element) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Create new helper for showing/hiding element

File: big-practice/src/js/pages/dashboard.page.js [hunk 4]
```diff
@@ -1,114 +1,115 @@
...
- */
-const showElement = (element) => {
-  element.classList.remove('hide');
-};
+  /**
+   * Show modal
+   *
+   * @param {element} element - Element of modal
+   */
+  showModal(element) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Create new class for showing/hiding a modal

File: big-practice/src/js/pages/dashboard.page.js [hunk 5]
```diff
@@ -123,163 +124,159 @@ const renderTable = () => {
...
+  // });
+
+  // // Movie will be updated when the update button is clicked
+  // modalFormUpdateBtn.addEventListener('click', () => {
+  //   handleUpdateForm();
+  //   hideModal(modalWarning);
+  // });
+
+  // Clicking on the logout button will log out of your account and return to the login page
+  logoutPage() {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Function name does not make sense. Should start with verb

File: big-practice/src/js/pages/dashboard.page.js (hunk 1 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 2 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Update!

File: big-practice/src/js/pages/dashboard.page.js (hunk 3 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 4 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 5 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/login.page.js [hunk 6]
```diff
@@ -89,6 +89,7 @@ class Login {
   checkUserLogin() {
     if (localStorage.getItem(USERNAME_KEY)) {
       window.location.href = DASHBOARD_PAGE;
+      return;
```
Comment (hieungo-agilityio) [inline big-practice/src/js/pages/login.page.js]: why did you need to return here?

File: big-practice/src/js/service/api.service.js [hunk 7]
```diff
@@ -49,7 +49,7 @@ class ApiService {
   /**
    * Remove by calling API
    */
-  async delete(url) {
+  async remove(url) {
```
Comment (hieungo-agilityio) [inline big-practice/src/js/service/api.service.js]: why did you rename function to remove?

File: big-practice/src/js/helpers/showing-modal.helper.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+export class ShowingModal {
+  /**
+   * Show modal
+   *
+   * @param {element} element - Element of modal
```
Comment (hieungo-agilityio) [inline big-practice/src/js/helpers/showing-modal.helper.js]: please double check type of `element` param.

File: big-practice/src/js/pages/dashboard.page.js [hunk 9]
```diff
@@ -1,114 +1,72 @@
...
+  //   showModal(modalWarning);
+  //   modalWarningDeleteBtn.setAttribute('data-id', movieId);
+  // }
+
+  /**
+   * Takes data from the API and displays it on a table in HTML
+   */
+  async renderTable() {
+    try {
+      const res = await fetch(MOVIES_API);
```
Comment (hieungo-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: why did you use fetch()?

File: big-practice/src/js/pages/dashboard.page.js [hunk 10]
```diff
@@ -123,163 +81,158 @@ const renderTable = () => {
...
 
-  getApi(`${MOVIES_API}?name=${data.name}`, (movieList) => {
-    const moviesDoNotExist = movieList.length === 0 || movieList[0].id === parseInt(formMovieId, 10);
-    
-    if (moviesDoNotExist) {
-      putApi(`${MOVIES_API}/${formMovieId}`, data, () => { renderTable(); });
-      hideModal(modalForm);
-    } else {
-      showErrorMessage(nameMovieInput, MESSAGES.exist);
+  checkUserLogin() {
```
Comment (hieungo-agilityio) [inline big-practice/src/js/pages/dashboard.page.js:217]: add comment.

File: big-practice/src/js/pages/dashboard.page.js [hunk 11]
```diff
@@ -123,163 +81,158 @@ const renderTable = () => {
...
+    // Allow return to login page if localStorage has no data
+    if (!localStorage.getItem(USERNAME_KEY)) {
+      window.location.href = LOGIN_PAGE;
+      return;
```
Comment (hieungo-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: do we need to return?

File: big-practice/src/js/pages/dashboard.page.js [hunk 12]
```diff
@@ -123,163 +81,158 @@ const renderTable = () => {
...
-  // Switch to login page
-  window.location.href = loginPage;
-});
+  }
 
-// Allow return to login page if localStorage has no data
-if (!localStorage.getItem('username')) {
-  window.location.href = loginPage;
-  return;
+  showUserName() {
```
Comment (hieungo-agilityio) [inline big-practice/src/js/pages/dashboard.page.js:227]: add comment.

File: big-practice/src/js/pages/dashboard.page.js [hunk 13]
```diff
@@ -123,163 +81,158 @@ const renderTable = () => {
...
+  //   hideModal(modalWarning);
+  // });
+
+  // // Movie will be updated when the update button is clicked
+  // modalFormUpdateBtn.addEventListener('click', () => {
+  //   handleUpdateForm();
+  //   hideModal(modalWarning);
+  // });
+
+  // Clicking on the logout button will log out of your account and return to the login page
```
Comment (hieungo-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: this is not comment style for function.

File: big-practice/src/js/pages/login.page.js (hunk 6 above)
Comment (jinety) [reply big-practice/src/js/pages/login.page.js]: Updated!

File: big-practice/src/js/service/api.service.js (hunk 7 above)
Comment (jinety) [reply big-practice/src/js/service/api.service.js]: Updated!

File: big-practice/src/js/helpers/showing-modal.helper.js (hunk 8 above)
Comment (jinety) [reply big-practice/src/js/helpers/showing-modal.helper.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 9 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 10 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js:217]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 11 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 13 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 12 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js:227]: Updated!

File: big-practice/src/js/helpers/document.helper.js [hunk 14]
```diff
@@ -10,4 +12,31 @@ export class DocumentHelper {
...
+    * @param {HtmlInputElement} element - Element input
+    */
+  cleanErrorMessage(element) {
+    this.showErrorMessage(element, EMPTY_TEXT);
+  }
+
+  /**
+   * Hide element
+   *
+   * @param {ElementCSSInlineStyle} element - Element
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/helpers/document.helper.js]: ElementCSSInlineStyle??

File: big-practice/src/js/pages/dashboard.page.js [hunk 15]
```diff
@@ -1,114 +1,75 @@
...
+   */
+  async renderTable() {
+    try {
+      const result = await apiService.get(MOVIES_API);
       let tableTemplate = '';
 
-      movies.forEach((movie) => {
-        tableTemplate += ` 
+      result.forEach((movie) => {
+        tableTemplate += `
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/pages/dashboard.page.js]: Split this code to a template as function. So we can use it as below:

tableTemplate += MovieTemplate.renderTable(move);

File: big-practice/src/js/templates/movie.template.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
+export class MovieTemplate {
+  static renderTable(move) {
```
Comment (phuocduong-agilityio) [inline big-practice/src/js/templates/movie.template.js]: Should be renderTableRow? Btw what does `test` mean ??

File: big-practice/src/js/helpers/document.helper.js (hunk 14 above)
Comment (jinety) [reply big-practice/src/js/helpers/document.helper.js]: Updated!

File: big-practice/src/js/pages/dashboard.page.js (hunk 15 above)
Comment (jinety) [reply big-practice/src/js/pages/dashboard.page.js]: Updated!

File: big-practice/src/js/templates/movie.template.js (hunk 16 above)
Comment (jinety) [reply big-practice/src/js/templates/movie.template.js]: Updated!
